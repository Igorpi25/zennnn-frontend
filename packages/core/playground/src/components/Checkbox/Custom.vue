<template>
  <div class="space-y-6">
    <Checkbox
      label="Red"
      input-class="peer-checked:bg-green-500 peer-checked:border-green-500 peer-focus:ring-green-400"
    >
      <span class="ml-2">Green color</span>
    </Checkbox>
    <Checkbox
      input-class="peer-checked:bg-red-600 peer-checked:border-red-600 peer-focus:ring-red-500 w-6 h-6"
      control-class="items-center"
    >
      <span class="ml-2">Red color and size</span>
    </Checkbox>
    <Checkbox :checked-icon="ziStatusPoint">
      <span class="ml-2">Custom checked icon</span>
    </Checkbox>
  </div>
</template>

<script setup>
import { ziStatusPoint } from '@zennnn/icons'
</script>
