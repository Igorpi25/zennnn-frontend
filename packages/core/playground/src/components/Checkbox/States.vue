<template>
  <div class="grid grid-flow-col">
    <Checkbox :model-value="true" label="on">
    </Checkbox>
    <Checkbox :model-value="false" label="off">
    </Checkbox>
    <Checkbox indeterminate label="indeterminate">
    </Checkbox>
  </div>
  <div class="grid grid-flow-col">
    <Checkbox :model-value="true" label="on disabled" disabled>
    </Checkbox>
    <Checkbox :model-value="false" label="off disabled" disabled>
    </Checkbox>
    <Checkbox indeterminate label="indeterminate disabled" disabled>
    </Checkbox>
  </div>
</template>

<script setup>
import { ref } from 'vue'

const model = ref(false)
</script>
