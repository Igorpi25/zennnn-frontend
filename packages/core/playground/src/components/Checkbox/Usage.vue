<template>
  <Checkbox v-model="model">
    <span class="ml-2">Value: {{ model }}</span>
  </Checkbox>
</template>

<script setup>
import { ref } from 'vue'

const model = ref(false)
</script>
