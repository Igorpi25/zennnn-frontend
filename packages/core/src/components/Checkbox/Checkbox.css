.checkbox--disabled {
  @apply opacity-50;
}

.checkbox__control {
  @apply flex items-start;
}
.checkbox__control--show-details {
  @apply pb-2;
}

.checkbox__icon {
  @apply w-5 h-5 border-2 inline-flex items-center justify-center rounded cursor-pointer;
  @apply transition-colors duration-100 ease-in-out;
  @apply border-gray-300 text-white;
  @apply peer-focus:ring peer-focus:ring-blue-400;
  @apply peer-checked:border-blue-500 peer-checked:bg-blue-500;
  @apply peer-disabled:border-gray-300 peer-disabled:peer-checked:bg-gray-300 peer-disabled:cursor-not-allowed;
  @apply peer-indeterminate:border-gray-300 peer-indeterminate:bg-gray-300;
}

.checkbox input:focus:not(:focus-visible) + .checkbox__icon {
  box-shadow: none;
  outline: 0;
}

.checkbox__label {
  @apply select-none leading-tight cursor-pointer;
  @apply peer-disabled:cursor-not-allowed;
}
