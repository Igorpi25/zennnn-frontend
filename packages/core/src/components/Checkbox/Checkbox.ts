import {
  h,
  ref,
  computed,
  watch,
  nextTick,
  vShow,
  withDirectives,
  defineComponent,
} from 'vue'
import { deepEqual } from 'vue-supp'

import { ziCheckboxMarked, ziMinus } from '@zennnn/icons'

import { useInputProps, useInput } from '../../composables/useInput'
import { useInputValidationProps, useInputValidation } from '../../composables/useInputValidation'

import uid from '../../utils/uid'

import Icon from '../Icon'

export default defineComponent({
  name: 'Checkbox',

  props: {
    ...useInputProps(),
    ...useInputValidationProps(),
    modelValue: {
      type: [Boolean, String, Number, Array],
      default: false,
    },
    value: {
      type: [Boolean, String, Number],
      default: false,
    },
    indeterminate: {
      type: Boolean,
      defalt: false,
    },
    multiple: {
      type: Boolean,
      default: null,
    },
    valueComparator: {
      type: Function,
      default: deepEqual,
    },
    trueValue: null,
    falseValue: null,
  },

  emits: ['update:modelValue', 'update:indeterminate', 'update:error', 'change'],

  setup (props, { slots, emit }) {
    const id: string = uid('checkbox-')
    const rootElement = ref<HTMLElement>()

    const {
      internalValue,
      isFocused,
      genInput,
      genLabel,
    } = useInput(props, { slots, id })

    const {
      showDetails,
      isDisabled,
      isReadonly,
      isInteractive,
      genMessages,
    } = useInputValidation(props, { emit, id, internalValue, isFocused })

    const inputIndeterminate = ref<boolean>(props.indeterminate)

    const isActive = computed(() => {
      const value = props.value
      const input = internalValue.value

      if (isMultiple.value) {
        if (!Array.isArray(input)) return false

        return input.some(item => props.valueComparator(item, value))
      }

      if (props.trueValue === undefined || props.falseValue === undefined) {
        return value
          ? props.valueComparator(value, input)
          : Boolean(input)
      }

      return props.valueComparator(input, props.trueValue)
    })

    const isMultiple = computed(() => props.multiple === true ||
      (props.multiple === null && Array.isArray(internalValue.value)))

    watch(() => props.modelValue, (val) => {
      internalValue.value = val
    })

    watch(() => props.indeterminate, (val) => {
      nextTick(() => {
        inputIndeterminate.value = val
      })
    })

    watch(inputIndeterminate, (val) => {
      emit('update:indeterminate', val)
    })

    watch(isActive, () => {
      if (!props.indeterminate) return
      inputIndeterminate.value = false
    })

    watch(internalValue, (val) => {
      emit('update:modelValue', val)
    })

    const onChange = () => {
      if (!isInteractive.value) return

      const value = props.value
      let input = internalValue.value

      if (isMultiple.value) {
        if (!Array.isArray(input)) {
          input = []
        }

        const length = input.length

        input = input.filter((item: any) => !props.valueComparator(item, value))

        if (input.length === length) {
          input.push(value)
        }
      } else if (props.trueValue !== undefined && props.falseValue !== undefined) {
        input = props.valueComparator(input, props.trueValue) ? props.falseValue : props.trueValue
      } else if (value) {
        input = props.valueComparator(input, value) ? null : value
      } else {
        input = !input
      }

      internalValue.value = input
      emit('change', internalValue.value)
    }

    const genCheckboxLabel = () => {
      const children = slots.default?.()
      if (!children) return undefined
      return h('label', {
        for: id,
        class: 'checkbox__label',
      }, children)
    }

    const genCheckboxIcon = () => {
      return h('div', {
        class: 'checkbox__control__icon',
      }, withDirectives(
        h(Icon, {
          size: 20,
        }, {
          default: () => inputIndeterminate.value ? ziMinus : ziCheckboxMarked,
        }),
        [
          [vShow, isActive.value || inputIndeterminate.value],
        ],
      ))
    }

    const genCheckboxInput = () => {
      const data = {
        id,
        class: 'checkbox__input',
        value: props.value,
        checked: isActive.value,
        type: 'checkbox',
        role: 'checkbox',
        readonly: isReadonly.value,
        disabled: isDisabled.value,
        'aria-checked': inputIndeterminate.value
          ? 'mixed'
          : isActive.value,
        onChange: onChange,
      }
      return genInput(data)
    }

    const genCheckbox = () => {
      return h('div', {
        class: 'checkbox__control__input',
      }, [
        genCheckboxInput(),
        genCheckboxIcon(),
      ])
    }

    const genControl = () => {
      return h('div', {
        class: {
          checkbox__control: true,
        },
      }, [
        genCheckbox(),
        genCheckboxLabel(),
      ])
    }

    return () => {
      return h('div', {
        ref: rootElement,
        class: {
          checkbox: true,
          'checkbox--active': isActive.value,
          'checkbox--focused': isFocused.value,
          'checkbox--disabled': isDisabled.value,
          'checkbox--readonly': isReadonly.value,
          'checkbox--indeterminate': inputIndeterminate.value,
          'checkbox--show-details': showDetails.value,
        },
      }, [
        genLabel(),
        genControl(),
        genMessages(),
      ])
    }
  },
})
