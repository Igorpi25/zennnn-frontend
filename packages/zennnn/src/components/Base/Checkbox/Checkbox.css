@layer components {
  .checkbox__control {
    @apply flex items-start relative;
  }
  .checkbox--show-details .checkbox__control {
    @apply pb-2;
  }
  .checkbox__control__input {
    @apply relative;
  }
  .checkbox__control__icon {
    @apply w-5 h-5 border-2 border-light-gray-900 dark:border-gray-300 rounded-3;
    @apply flex items-center justify-center;
    @apply transition-colors duration-100 ease-in-out;
    @apply text-white;
  }
  .checkbox--focused .checkbox__control__icon {
    @apply ring ring-blue-500 ring-opacity-50;
  }
  .checkbox--active .checkbox__control__icon {
    @apply border-blue-500 bg-blue-500;
  }
  .checkbox--active.checkbox--disabled .checkbox__control__icon,
  .checkbox--indeterminate .checkbox__control__icon {
    @apply bg-light-gray-900 border-light-gray-900 dark:bg-gray-300 dark:border-gray-300;
  }

  .checkbox__input {
    @apply absolute opacity-0 select-none w-full h-full;
  }

  .checkbox__label {
    @apply select-none leading-tight;
  }

  .checkbox__label,
  .checkbox__input {
    @apply cursor-pointer;
  }
  .checkbox--disabled .checkbox__label,
  .checkbox--disabled .checkbox__input {
    @apply cursor-not-allowed;
  }

  .checkbox--disabled {
    @apply opacity-40;
  }
}
