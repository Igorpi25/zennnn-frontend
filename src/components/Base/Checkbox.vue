<template>
  <InputBase
    :is-dirty="!!internalValue"
    :has-error="hasError"
    :hide-details="hideDetails"
    :detail-text="errorText"
    :class="[
      'checkbox',
      {'checkbox--secondary': secondary},
      {'checkbox--vertical': verticalAlign},
      {'checkbox--horizontal': horizontalAlign},
    ]"
  >
    <div class="checkbox__slot">
      <div class="checkbox__input">
        <input
          ref="input"
          type="checkbox"
          v-model="internalValue"
          :id="inputId"
          :name="name"
          :required="required"
          :readonly="readonly"
          :disabled="disabled"
          @input="checkField"
        >
        <i class="checkbox__icon">
          <div class="icon__item">
            <svg v-if="checked" xmlns="http://www.w3.org/2000/svg" width="14" height="14" viewBox="0 0 14 14"><title>checkbox</title><path d="M7,3a4,4,0,1,0,4,4A4,4,0,0,0,7,3Zm0,7a3,3,0,1,1,3-3A3,3,0,0,1,7,10Z" style="fill:#5a8199"/><circle cx="7" cy="7" r="3" style="fill:#16a0ce"/><path d="M0,0V14H14V0H0ZM12,12H2V2H12V12Z" style="fill:currentColor"/></svg>
            <svg v-else xmlns="http://www.w3.org/2000/svg" width="14" height="14" viewBox="0 0 14 14"><title>checkbox-empty</title><path d="M0,0V14H14V0H0ZM12,12H2V2H12V12Z" style="fill:currentColor"/></svg>
          </div>
        </i>
      </div>
      <label :for="inputId" class="cursor-pointer">
        <slot />
      </label>
    </div>
  </InputBase>
</template>

<script>
import validatable from '@/mixins/validatable'

export default {
  name: 'Checkbox',
  inject: {
    form: {
      default: null,
    },
  },
  mixins: [validatable],
  props: {
    value: {
      type: Boolean,
      default: false,
    },
    rules: {
      type: Array,
    },
    label: {
      type: String,
      default: '',
    },
    name: {
      type: String,
      default: null,
    },
    required: {
      type: Boolean,
      default: false,
    },
    readonly: {
      type: Boolean,
      default: false,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    hideDetails: {
      type: Boolean,
      default: false,
    },
    horizontalAlign: {
      type: Boolean,
      default: false,
    },
    verticalAlign: {
      type: Boolean,
      default: false,
    },
    secondary: {
      type: Boolean,
      default: false,
    },
  },
  data () {
    return {
      checked: false,
      // TODO input registrator
      inputId: 'input' + Math.round(Math.random() * 100000),
      lazyValue: this.value,
    }
  },
  computed: {
    internalValue: {
      get () {
        return this.lazyValue
      },
      set (val) {
        this.lazyValue = val
        this.checked = !!val
        this.$emit('input', val)
      },
    },
  },
  watch: {
    value: {
      handler (val) {
        this.lazyValue = val
        this.checked = !!val
      },
      immediate: true,
    },
  },
  created () {
    if (this.form) {
      this.form.register(this)
    }
  },
  beforeDestroy () {
    if (this.form) {
      this.form.unregister(this)
    }
  },
}
</script>
