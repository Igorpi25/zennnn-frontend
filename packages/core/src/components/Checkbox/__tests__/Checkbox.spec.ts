import { mount, VueWrapper } from '@vue/test-utils'

import Checkbox from '../Checkbox'

jest.mock('../../../utils/uid')

describe('Checkbox.ts', () => {
  let mountFunction: (options?: Record<string, unknown>) => VueWrapper<any>

  beforeEach(() => {
    mountFunction = (options = {}) => {
      return mount(Checkbox, {
        ...options,
      })
    }
  })

  // Render

  it('should render component and match snapshot', () => {
    const wrapper = mountFunction({
      slots: {
        default: () => 'Label',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with disabled prop and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        disabled: true,
      },
      slots: {
        default: () => 'Label',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with indeterminate prop and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        indeterminate: true,
      },
      slots: {
        default: () => 'Label',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with name, required, readonly props and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        name: 'simple-input',
        required: true,
        readonly: true,
      },
      slots: {
        default: () => 'Label',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render checked component and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: true,
      },
      slots: {
        default: () => 'Label',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render focused component and match snapshot', async () => {
    const wrapper = mountFunction({
      slots: {
        default: () => 'Label',
      },
    })

    await wrapper.get('input').trigger('focus')

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with label prop and match snapshot', async () => {
    const wrapper = mountFunction({
      props: {
        label: 'Input label',
      },
      slots: {
        default: () => 'Checkbox label',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  // ---

  it('should emit true/false when input/label clicked', async () => {
    const wrapper = mountFunction({
      slots: {
        default: () => 'Label',
      },
    })

    const input = wrapper.get('input')
    await input.setValue('false')
    await input.setValue('true')

    // on click emitted native click
    await input.trigger('click')
    await wrapper.get('label').trigger('click')
    expect(wrapper.emitted().click.length).toEqual(3)

    expect(wrapper.emitted()).toHaveProperty('change')
    expect(wrapper.emitted().change[0]).toEqual([true])
    expect(wrapper.emitted().change[1]).toEqual([false])
  })

  it('should emit specified value and return null on toggled off when clicked', async () => {
    const wrapper = mountFunction({
      props: {
        value: 'Test',
      },
    })

    const input = wrapper.get('input')
    await input.setValue()
    await input.setValue('Test')

    // on click emitted native click
    expect(wrapper.emitted().change[0]).toEqual(['Test'])
    expect(wrapper.emitted().change[1]).toEqual([null])
  })

  it('should work with custom true- and false-value', async () => {
    const wrapper = mountFunction({
      props: {
        trueValue: 'on',
        falseValue: 'off',
      },
    })

    const input = wrapper.get('input')
    await input.setValue('false')
    await input.setValue('true')

    expect(wrapper.emitted().change[0]).toEqual(['on'])
    expect(wrapper.emitted().change[1]).toEqual(['off'])
  })

  it('should emit array value on multiple prop', async () => {
    const wrapper = mountFunction({
      props: {
        value: 'Test',
        multiple: true,
      },
    })

    await wrapper.get('input').setValue()

    expect(wrapper.emitted().change[0]).toEqual([['Test']])
  })

  it('should emit array value with modelValue is array', async () => {
    const wrapper = mountFunction({
      props: {
        modelValue: [],
        value: 'Test',
      },
    })

    await wrapper.get('input').setValue()

    expect(wrapper.emitted().change[0]).toEqual([['Test']])
  })

  it('should toggle value in array on multiple prop', async () => {
    const wrapper = mountFunction({
      props: {
        modelValue: ['Test'],
        value: 'Test',
        multiple: true,
      },
    })

    await wrapper.get('input').setValue(['Test'])

    expect(wrapper.emitted().change[0]).toEqual([[]])
  })
})
