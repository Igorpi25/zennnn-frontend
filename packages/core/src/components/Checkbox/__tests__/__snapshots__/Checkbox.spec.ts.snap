// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox.ts should render checked component and match snapshot 1`] = `
<div class="checkbox checkbox--active">
  <!---->
  <div class="checkbox__control">
    <div class="checkbox__control__input"><input id="checkbox-1" class="checkbox__input" type="checkbox" value="false" role="checkbox" aria-checked="true">
      <div class="checkbox__control__icon"><i class="icon" aria-hidden="true"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="20px" width="20px" role="img" aria-hidden="true" style="font-size: 20px; height: 20px; width: 20px;">
            <g fill="currentColor">
              <path d="m20.003 8.552-8.871 8.827L4.8 11.188l2.517-2.574 3.793 3.709L17.464 6l2.54 2.552z"></path>
            </g>
          </svg></i></div>
    </div><label for="checkbox-1" class="checkbox__label">Label</label>
  </div>
  <!---->
</div>
`;

exports[`Checkbox.ts should render component and match snapshot 1`] = `
<div class="checkbox">
  <!---->
  <div class="checkbox__control">
    <div class="checkbox__control__input"><input id="checkbox-1" class="checkbox__input" type="checkbox" value="false" role="checkbox" aria-checked="false">
      <div class="checkbox__control__icon"><i class="icon" aria-hidden="true" style="display: none;"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="20px" width="20px" role="img" aria-hidden="true" style="font-size: 20px; height: 20px; width: 20px;">
            <g fill="currentColor">
              <path d="m20.003 8.552-8.871 8.827L4.8 11.188l2.517-2.574 3.793 3.709L17.464 6l2.54 2.552z"></path>
            </g>
          </svg></i></div>
    </div><label for="checkbox-1" class="checkbox__label">Label</label>
  </div>
  <!---->
</div>
`;

exports[`Checkbox.ts should render component with disabled prop and match snapshot 1`] = `
<div class="checkbox checkbox--disabled">
  <!---->
  <div class="checkbox__control">
    <div class="checkbox__control__input"><input disabled="" id="checkbox-1" class="checkbox__input" type="checkbox" value="false" role="checkbox" aria-checked="false">
      <div class="checkbox__control__icon"><i class="icon" aria-hidden="true" style="display: none;"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="20px" width="20px" role="img" aria-hidden="true" style="font-size: 20px; height: 20px; width: 20px;">
            <g fill="currentColor">
              <path d="m20.003 8.552-8.871 8.827L4.8 11.188l2.517-2.574 3.793 3.709L17.464 6l2.54 2.552z"></path>
            </g>
          </svg></i></div>
    </div><label for="checkbox-1" class="checkbox__label">Label</label>
  </div>
  <!---->
</div>
`;

exports[`Checkbox.ts should render component with indeterminate prop and match snapshot 1`] = `
<div class="checkbox checkbox--indeterminate">
  <!---->
  <div class="checkbox__control">
    <div class="checkbox__control__input"><input id="checkbox-1" class="checkbox__input" type="checkbox" value="false" role="checkbox" aria-checked="mixed">
      <div class="checkbox__control__icon"><i class="icon" aria-hidden="true"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="20px" width="20px" role="img" aria-hidden="true" style="font-size: 20px; height: 20px; width: 20px;">
            <g fill="currentColor">
              <path d="M7.2 10.8h9.6v2.4H7.2v-2.4z"></path>
            </g>
          </svg></i></div>
    </div><label for="checkbox-1" class="checkbox__label">Label</label>
  </div>
  <!---->
</div>
`;

exports[`Checkbox.ts should render component with label prop and match snapshot 1`] = `
<div class="checkbox"><label id="label-1" for="checkbox-1" aria-hidden="false" class="label">
    <div class="label__wrapper">
      <div class="label__content__wrapper">
        <div class="label__content">Input label</div>
      </div>
      <!---->
    </div>
  </label>
  <div class="checkbox__control">
    <div class="checkbox__control__input"><input id="checkbox-1" class="checkbox__input" type="checkbox" value="false" role="checkbox" aria-checked="false">
      <div class="checkbox__control__icon"><i class="icon" aria-hidden="true" style="display: none;"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="20px" width="20px" role="img" aria-hidden="true" style="font-size: 20px; height: 20px; width: 20px;">
            <g fill="currentColor">
              <path d="m20.003 8.552-8.871 8.827L4.8 11.188l2.517-2.574 3.793 3.709L17.464 6l2.54 2.552z"></path>
            </g>
          </svg></i></div>
    </div><label for="checkbox-1" class="checkbox__label">Checkbox label</label>
  </div>
  <!---->
</div>
`;

exports[`Checkbox.ts should render component with name, required, readonly props and match snapshot 1`] = `
<div class="checkbox checkbox--readonly">
  <!---->
  <div class="checkbox__control">
    <div class="checkbox__control__input"><input name="simple-input" required="" readonly="" id="checkbox-1" class="checkbox__input" type="checkbox" value="false" role="checkbox" aria-checked="false">
      <div class="checkbox__control__icon"><i class="icon" aria-hidden="true" style="display: none;"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="20px" width="20px" role="img" aria-hidden="true" style="font-size: 20px; height: 20px; width: 20px;">
            <g fill="currentColor">
              <path d="m20.003 8.552-8.871 8.827L4.8 11.188l2.517-2.574 3.793 3.709L17.464 6l2.54 2.552z"></path>
            </g>
          </svg></i></div>
    </div><label for="checkbox-1" class="checkbox__label">Label</label>
  </div>
  <!---->
</div>
`;

exports[`Checkbox.ts should render focused component and match snapshot 1`] = `
<div class="checkbox checkbox--focused">
  <!---->
  <div class="checkbox__control">
    <div class="checkbox__control__input"><input id="checkbox-1" class="checkbox__input" type="checkbox" value="false" role="checkbox" aria-checked="false">
      <div class="checkbox__control__icon"><i class="icon" aria-hidden="true" style="display: none;"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="20px" width="20px" role="img" aria-hidden="true" style="font-size: 20px; height: 20px; width: 20px;">
            <g fill="currentColor">
              <path d="m20.003 8.552-8.871 8.827L4.8 11.188l2.517-2.574 3.793 3.709L17.464 6l2.54 2.552z"></path>
            </g>
          </svg></i></div>
    </div><label for="checkbox-1" class="checkbox__label">Label</label>
  </div>
  <!---->
</div>
`;
